tests/unit/s2n_security_policies_test.c

```diff
@@ -395,6 +397,40 @@ int main(int argc, char **argv)
             }
         }
     }
+
+    /* Certificate preferences list needs to allow all the rsa pss signature schemes as we currently don't differentiate
+     * between these schemes.
+     */
+    {
+        const struct s2n_signature_scheme* const rsa_pss_sig_scheme_list[] = {
+            &s2n_rsa_pss_pss_sha256,
+            &s2n_rsa_pss_pss_sha384,
+            &s2n_rsa_pss_pss_sha512,
+            &s2n_rsa_pss_rsae_sha256,
+            &s2n_rsa_pss_rsae_sha384,
+            &s2n_rsa_pss_rsae_sha512,
+        };
+
+        for (size_t i = 0; security_policy_selection[i].version != NULL; i++) {
+            security_policy = security_policy_selection[i].security_policy;
+            EXPECT_NOT_NULL(security_policy);
+
+            if (security_policy->certificate_signature_preferences != NULL) {
+
+                for (size_t j = 0; j < s2n_array_len(rsa_pss_sig_scheme_list); j++) {
+                    const struct s2n_signature_scheme *scheme = rsa_pss_sig_scheme_list[j];
+                    bool contains = false;
+                    for (size_t k = 0; k < security_policy->certificate_signature_preferences->count; k++) {
+                        if (scheme == security_policy->certificate_signature_preferences->signature_schemes[k]) {
+                            contains = true;
+                            break;
+                        }
+                    }
+                    EXPECT_TRUE(contains);
```

Comment:
Doesn't this verify that every certificate_signature_preferences contains at least one PSS algorithm? What you actually need to verify is that if a certificate_signature_preferences contains any PSS algorithm, then it contains them all, because we can't tell them apart.

To make sure you're doing this right, you'll probably want to make sure you test it's false for at least one example. Maybe put the actual check in s2n_security_policies_init, and just verify that it works here.

Reply:
I’m pretty sure this is testing that every existing certificate_signatures_preference has all 6 rsa_pss algorithms allowed. But I can rewrite it so that it checks that if a cert_sig_preference has any PSS algorithm, then it has them all.

Reply:
Nope, you're right, I misunderstood this check.